core/dbt/include/global_project/macros/relations/materialized_view/drop.sql

```diff
@@ -11,4 +11,5 @@ actually executes the drop, and `get_drop_sql`, which returns the template.
 
 {% macro default__drop_materialized_view(relation) -%}
     drop materialized view if exists {{ relation }} cascade
+    {{ adapter.cache_dropped(relation) }}
```

Comment:
I think this needs to go in the main macro that dispatches here, and then remove it from the postgres versions below. It seems like we're missing a drop scenario based on what's happening below. i.e. we're dropping the object, not updating the cache, doing all the replace work, and then trying to rename back to the thing that we dropped (but which is still in the cache).

Reply:
Can do, I initially started pulling them into these calls because I worried the dispatch macro wasn't going to get stopped especially on `drop` as it is an actual return on the call